integration-test/src/camera_recorder_tests.rs

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+
+#[cfg(test)]
+mod test {
+    use gst::prelude::*;
+    use gst::ClockType::Realtime;
+    use pravega_video::timestamp::{PravegaTimestamp, SECOND};
+    use rstest::rstest;
+    #[allow(unused_imports)]
+    use tracing::{error, info, debug};
+    use uuid::Uuid;
+    use crate::*;
+    use crate::rtsp_camera_simulator::{start_or_get_rtsp_test_source, RTSPCameraSimulatorConfigBuilder};
+    use crate::utils::*;
+
+    #[rstest]
+    #[case(Realtime, 30, 15)]
+    fn test_ungraceful_stop(#[case] clock_type: gst::ClockType, #[case] num_sec_to_record: u64, #[case] num_sec_to_failure: u64) {
+        gst_init();
+        let clock = gst::SystemClock::obtain();
+        clock.set_property("clock-type", &clock_type).unwrap();
+        gst::SystemClock::set_default(Some(&clock));
+        let rtsp_server_config = RTSPCameraSimulatorConfigBuilder::default().fps(20).build().unwrap();
+        let num_frames = num_sec_to_record * &rtsp_server_config.fps;
+        let (rtsp_url, _rtsp_server) = start_or_get_rtsp_test_source(rtsp_server_config);
+        let test_config = &get_test_config();
+        info!("### BEGIN:test_config={:?}", test_config);
+        let stream_name = &format!("test-pravegasrc-{}-{}", test_config.test_id, Uuid::new_v4())[..];
+        let pipeline_description = format!("\
+            rtspsrc \
+              buffer-mode=none \
+              drop-on-latency=true \
+              latency=2000 \
+              location={rtsp_url} \
+              ntp-sync=true \
+              ntp-time-source=running-time \
+            ! rtph264depay \
+            ! h264parse \
+            ! video/x-h264,alignment=au \
+            ! identity name=h264par silent=false eos-after={num_frames} \
+            ! timestampcvt \
+            ! identity name=tscvt__ silent=false \
+            ! mp4mux streamable=true fragment-duration=1 ! fragmp4pay \
+            ! pravegasink {pravega_plugin_properties} \
+              timestamp-mode=tai sync=false \
+            ",
+            rtsp_url = rtsp_url,
+            num_frames = num_frames,
+            pravega_plugin_properties = test_config.pravega_plugin_properties(stream_name),
+        );
+        let expected_timestamp = PravegaTimestamp::now();
+        let _ = launch_pipeline_and_get_summary(format!("{} simulate-failure-after-sec={}", pipeline_description, num_sec_to_failure).as_ref());
+
+        info!("#### Read recorded stream from Pravega, no demux, no decoding, part 1");
+        let pipeline_description_read = format!(
```

Comment:
This should test that the video is also decodable. Use the code from https://github.com/pravega/gstreamer-pravega/blob/master/integration-test/src/rtsp_tests.rs#L200-L228. Some frames around the ungraceful stop may be corrupted but others before and after should be good.

Reply:
Added.